docker/dockerProxyServer.js

```diff
@@ -0,0 +1,303 @@
+    return this._getCurrentContainerId().then(({ containerId }) => {
+      return this._internalConnectToNetwork({ networkId, containerId });
+    });
+  }
+
+  // returns Promise<[{ Name, Id }]>
+  _getCurrentNetworks() {
+    return this._inspectCurrentContainer().then((currentContainer) => {
+      const networkNames = Object.keys(currentContainer.NetworkSettings.Networks);
+      return networkNames.map((networkName) => {
+        return {
+          Name: networkName,
+          Id: currentContainer.NetworkSettings.Networks[networkName].NetworkID
+        };
+      });
+    });
+  }
+
+  _disconnectFromAllNetworks() {
+    return this._getCurrentNetworks().then((networks) => {
+      const disconnectPromises = networks.map(({ Id }) => this._disconnectFromNetwork({ networkId: Id }));
+      return Promise.all(disconnectPromises);
+    });
+  }
+
+  _disconnectFromNetwork({ networkId }) {
+    return this._getCurrentContainerId().then(({ containerId }) => {
+      return this._internalDisconnectFromNetwork({ networkId, containerId });
+    });
+  }
+
+  _createNetwork({ networkName }) {
+    // Note: we tag the networks created by this instance of proxy
+    // with a label, so that we can prune specific networks during cleanup.
+    const instanceId = (new Date()).toDateString();
+    const postData = JSON.stringify({
+      'Name': networkName,
+      'Labels': {
+        'dockerProxy': instanceId,
+        [this._instanceLabel]: instanceId,
+      }
+    });
+    return this._makeRequest({
+      socketPath: '/var/run/docker.sock',
+      path: '/v1.32/networks/create',
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'Content-Length': postData.length
+      }
+    }, postData);
+  }
+
+  _isDocker() {
+    return Promise.resolve({ isDocker });
+  }
+
+  _version() {
+    return Promise.resolve({ version });
+  }
+
+  _internalConnectToNetwork({ networkId, containerId }) {
+    const postData = JSON.stringify({
+      'Container': containerId,
+    });
+
+    return this._makeRequest({
+      socketPath: '/var/run/docker.sock',
+      path: `/v1.32/networks/${networkId}/connect`,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'Content-Length': postData.length
+      }
+    }, postData);
+  }
+
+  _internalDisconnectFromNetwork({ networkId, containerId }) {
+    const postData = JSON.stringify({
+      'Container': containerId,
+      'Force': false
+    });
+    return this._makeRequest({
+      socketPath: '/var/run/docker.sock',
+      path: `/v1.32/networks/${networkId}/disconnect`,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'Content-Length': postData.length
+      }
+    }, postData);
+  }
+
+  _makeRequest(options, postdata) {
+    const thisRequest = this._requestId++;
+    return new Promise((resolve, reject) => {
+      let clientRequest = http.request(options, (res) => {
```

Comment:
@manjeshbhargav, I will fix this, but was hoping to find a way to auto-catch such issues so that they need not be part of the PR - there was a promising 'no-extra-parens' eslint directive, but unfortunately it does not catch this one.